hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -3617,8 +3617,12 @@ def __getitem__(self, item):
                         stop = hl.cond(step > 0, dlen, to_expr(-1, tint64))
 
                     slices.append(hl.tuple((start, stop, step)))
-                else:
-                    slices.append(s)
+                else:  # Not a slice, just an int index
+                    adjusted_index = hl.if_else(s < 0, s + dlen, s)
+                    checked_int = hl.case().when((adjusted_index < dlen) & (adjusted_index >= 0), adjusted_index).or_error(
+                        hl.str("Index ") + hl.str(s) + hl.str(f" is out of bounds for axis {i} with size ") + hl.str(dlen)
```

Comment:
This can be simplified to a single statement I think

```python
.or_error(f"Index {hl.str(s)} is out of bounds for axis {i} with size {hl.str(dlen)}
```

Reply:
You can't do this. `f` strings get turned into python strings at python runtime, which means you get:

```
Index <StringExpression of type str> is out of bounds for axis 0 with size <StringExpression of type str>'
```

Reply:
We'd need a special hail string formatting thing. Maybe that exists, I didn't look hard for it.

Reply:
yeah, cool, thanks for the explanation